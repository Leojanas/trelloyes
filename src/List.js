import React from 'react';
import Card from './Card.js';
import './List.css'

function List(props) {
    
    return(
        <section className='List'>
            <header className='List-header'>
                <h2>{props.header}</h2>
            </header>
            <div className='List-cards'>
                {props.cards.map((card) => 
                    <Card
                    id={card.id}
                    key={card.id}
                    title={card.title}
                    content={card.content}
                    onDeleteCard={props.onDeleteCard}
                    />
                )}
                <button type='button' className='List-add-button' onClick={() => props.onAddRandomCard(props.id)}>
                    + Add Random Card
                </button>
            </div>
        </section>
        
    )
}

export default List;