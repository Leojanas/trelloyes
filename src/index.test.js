import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import List from './List';
import Store from './store';
import renderer from 'react-test-renderer';

describe('Card component', ()=>{

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card title='title' content='content'/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<Card title='title' content='content'/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });

})

describe('List component', () => {
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key={1} header='First List' cards={[{ id: 'a', title: 'First card', content: 'lorem ipsum' },
        { id: 'b', title: 'Second card', content: 'lorem ipsum' }]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<List key={1} header='First List' cards={[{ id: 'a', title: 'First card', content: 'lorem ipsum' },
          { id: 'b', title: 'Second card', content: 'lorem ipsum' }]}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
})