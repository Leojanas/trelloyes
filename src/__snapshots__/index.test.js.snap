// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    Delete
  </button>
  <h3>
    title
  </h3>
  <p>
    content
  </p>
</div>
`;

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
